packages/react-reconciler/src/DebugTracing.js

```diff
@@ -17,6 +18,10 @@ let nativeConsoleLog: null | Function = null;
 const pendingGroupArgs: Array<any> = [];
 let printedGroupIndex: number = -1;
 
+function formatLanes(laneOrLanes: Lane | Lanes): string {
+  return '0b' + (laneOrLanes: any).toString(2).padStart(30, '0');
```

Comment:
We could do fancier formatting here (including showing the lane "names" based on which bit positions were flipped on, but that would be more work to keep in sync and may change frequently with the implementation details- so just printing the number is probably sufficient?)

Reply:
I think we shouldn't leak that the Lanes type is a bitmask to the DevTools profiler. We should treat it more like a thread ID. So if you started rendering with lanes `0b10011`, the profiler would understand that as "Started rendering lanes: 0, 1, and 5." The actual transport format could use a bitmask for performance, but the profiler wouldn't deal with bitmasks anywhere else.

But I think it's good to use a bitmask for the debug tracing feature, since it's only used by us and will eventually  be replaced by the full profiler. Just making a note for the next step.

Reply:
I think it should be `padStart(31, '0')` since 31 is the max number of lanes.

Reply:
> the profiler wouldn't deal with bitmasks anywhere else.

Or it could, for performance reasons, but it would be its own type that doesn't assume anything about the bitmask type used by React.

The idea is that React should be able to change the layout of the Lanes type without breaking DevTools. For example, React might reserve one of the bits for some non-Lane purpose. Not sure how likely that is, but it's not inconceivable.

Reply:
I think we'd still probably use a bitmask/number for serialization/transport, but we can use a separate opaque Flow type to avoid leaking the `Lane`/`Lanes` type directly.

> but it would be its own type that doesn't assume anything about the bitmask type used by React.

Right, other than the fact that it is a bitmask.

That being said, I think we may end up wanting to convey *some* information to the Profiler somehow (metadata or Flow types) if we want to be able to e.g. treat the sync lane specially in some way.

Have you seen my mockup (in the internal Core group) btw?